x/nexus/keeper/keeper.go

```diff
@@ -324,3 +360,35 @@ func (k Keeper) setLatestDepositAddress(ctx sdk.Context, recipient exported.Cros
 	key := depositAddrPrefix.Append(utils.LowerCaseKey(recipient.String()))
 	k.getStore(ctx).SetRaw(key, []byte(address))
 }
+
+// merges cross chain transfers grouped by the given function
+func mergeTransfersBy(transfers []exported.CrossChainTransfer, groupFn func(transfer exported.CrossChainTransfer) string) []exported.CrossChainTransfer {
```

Comment:
This function should be removed once the above is changed according to my comments.

Reply:
done